### PR TITLE
[CDAP-8430] New "No Entities Found" message

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-8430

Display different messages and suggestions depending on the search query/filters applied.

### CODE REVIEW

**Comment by ajainarayanan** [comment]
@tonybach Noticed couple of things,
1. The font-size for the heading seems to be defined as 18px in the JIRA but it was 20px in the UI
2. I don't think the message is centered on the screen. 
3. When user changes the filter and if there are no entities then the message should be `No results found for your selection.` instead saw `No entities found for these filters`

Also not sure how we should handle this but when I end up in 3 step mentioned above I get the message for "no entities for your selection" message but when I start with a search query and then select my filters, I am getting the same "no entities for your selection" message. Should we indicate that for the current search and for the current filter selection there are no matches? May be @bdmogal or Lea can provide some inputs.

File: cdap-ui/app/cdap/components/EntityListView/ListView/index.js [hunk 1]
```diff
@@ -42,6 +47,36 @@ export default class HomeListView extends Component {
...
+        activeEntity: activeEntity[0]
+      });
+    }
+    if (this.props.onEntityClick) {
+      this.props.onEntityClick(entity);
+    }
+  }
+
+  filtersApplied() {
+    return this.props.activeFilter.length > 0 && this.props.activeFilter.length < this.props.filterOptions.length;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/EntityListView/ListView/index.js]: Typing in a search term with 0 results could show a message but unchecking even one option from the dropdown will show another message right? Might not be consistent

**Comment by ajainarayanan** [comment]
@tonybach Would be nice if we can have a unit test that confirms the spec in the JIRA. Easier to verify things.

File: cdap-ui/app/cdap/components/EntityListView/NoEntitiesMessage/__tests__/NoEntitiesMessage.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        searchText='abc'
+        filtersAreApplied={filtersAreApplied}
+      />
+    );
+    expect(noEntitiesMessage.find('.empty-message-container strong').text()).toBe(searchEmptyMessagePath);
+    expect(noEntitiesMessage.find('.empty-message-container .action-item').length).toBe(2);
+    noEntitiesMessage.unmount();
+  });
+
+  it('Should open up the Add Entity modal when "Add" is clicked', () => {
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/EntityListView/NoEntitiesMessage/__tests__/NoEntitiesMessage.test.js]: If time permits we could add one more test that will create `NoEntitiesMessage` with `onSearch, onFiltersCleared ` function that gets called on clicking the Clear filter link.

**Review by ajainarayanan — APPROVED** [review]
👍 Just a minor comment. LGTM 🎆

**Review by ajainarayanan — APPROVED** [review]
👍 LGTM

**Comment by tonybach** [comment]
Build was green: http://builds.cask.co/browse/CDAP-DRC5978

**Pull request merged**